# General Settings
omp_num_threads: 8

sogmm_node:
  # Logging
  log_level: "DEBUG"  # Options: DEBUG, INFO, WARN, ERROR, FATAL
  
  # Hardware Selection
  use_gpu: true  # Options: true (GPU), false (CPU)
  
  # SOGMM Algorithm Parameters
  bandwidth: 0.02
  tolerance: 1.0e-10
  reg_covar: 1.0e-8
  max_iter: 100
  kl_div_match_thresh: 100.0

  # Visualization Parameters
  enable_visualization: true
  visualization_scale: 2.0
  color_by: "confidence" # Options: intensity, confidence, stability, combined
  add_metric_text: false
  suspicious_displacement: 0.001
  unstable_displacement: 0.4
  displacement_score_factor: 0.15

  uncertainty_heuristic: "confidence" # Options: intensity, confidence, stability, combined
  uncertainty_scaler: 10.0

  # Processing Parameters
  processing_decimation: 1
  min_novel_points: 500
  target_frame: "map"

  # Pruning Parameters
  max_fusion_ratio: 0.4
  prune_min_observations: 5
  prune_interval_frames: 5

  # Freezing Parameters
  enable_freeze: false
  freeze_interval_frames: 5
  freeze_fusion_threshold: 20

  fusion_weight_update: true

  publish_whole_gmm: true

# (x,y) bounds
map_bounds: [[-0.8, 8.7], [-2.75, 2.9], [0.0, 2.0]] # hidden obstacle
# map_bounds: [[-3.5, 11.25], [-6., 5.8], [0.0, 3.0]] # big empty arena
# map_bounds: [[-0.65, 9.0], [-1.0, 4.5], [-0.7, 1.5]]  # custom arena (cluttered)

sogmm_grid_node:
  grid_cell_size: 2.0
  grid_marker_scale: 0.5
  grid_offset: 1.0
  unexplored_uncertainty: 1.0

sogmm_exploration_node:
  exploration_distance_gain: 0.2
  clamp_z: True
  distance_threshold: 1.0

simple_exploration: False 